providers/nitro/nitro-helper/src/tracing_layer.rs

```diff
@@ -294,10 +319,18 @@ impl Log {
 
                 let mut value_len_raw = [0; 8];
                 value_len_raw.copy_from_slice(&raw[index_value_len_begin..index_value_len_end]);
-                let value_len = u64::from_le_bytes(value_len_raw) as usize;
+                let value_len_u64 = u64::from_le_bytes(value_len_raw);
+                if value_len_u64 > usize::MAX as u64 {
+                    warn!("unable to convert slice index from u64 to usize when value_len_raw in u64 > {}", usize::MAX);
+                    break;
+                }
 
                 let index_value_begin = index_value_len_end;
-                let index_value_end = index_value_begin + value_len;
+                let index_value_end = break_from_checked_add_overflow!(index_value_begin.checked_add(value_len_u64 as usize));
+                if index_value_end > len {
+                    break;
+                }
+
                 flag_index_next = index_value_end + 1;
```

Comment:
break_from_checked_add_overflow ?

Reply:
Even index_value_end is not overflowed, it could be bigger than len when value_len_raw is unusaully large?

Reply:
```
&raw[index_value_begin..index_value_end]
```
may panic if index_value_end is bigger than len?

Reply:
I'd need to try... accessing an element on an invalid index panics, but index ranges/slices may be ok as long as begin<=end

Reply:
@allthatjazzleo I just tried -- it seems it'd panic with: `range end index X out of range for slice of length L`